src/main/cpp/src/hive_hash.cu

```diff
@@ -210,11 +210,72 @@ class hive_device_row_hasher {
       return this->hash_functor.template operator()<T>(col, row_index);
     }
 
+    struct StackElement{
+      cudf::column_device_view col;   // current column
+      int child_idx;                  // index of the child column to process next
+      int factor_exp;                 // factor exponent for the current column
+      //factor_exp = parent.factor_exp + parent.child_num - 1 - parent.child_idx
+
+      __device__ StackElement() = delete;
+      __device__ StackElement(cudf::column_device_view col, int factor_exp) : col(col), child_idx(0), factor_exp(factor_exp) {}
+    };
+
+    typedef StackElement* StackElementPtr;
     template <typename T, CUDF_ENABLE_IF(cudf::is_nested<T>())>
     __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
                                             cudf::size_type row_index) const noexcept
     {
-      CUDF_UNREACHABLE("Nested type is not supported");
+      hive_hash_value_t ret = HIVE_INIT_HASH;
+      cudf::column_device_view curr_col = col.slice(row_index, 1);
+      // column_device_view default constructor is deleted, can not allocate column_device_view array directly
+      // use byte array to wrapper StackElement list
+      constexpr int len_of_maxlen_stack_element = MAX_NESTED_LEN * sizeof(StackElement);
+      uint8_t stack_wrapper[len_of_maxlen_stack_element];
+      StackElementPtr stack = reinterpret_cast<StackElementPtr>(stack_wrapper);
+      int stack_size = 0;
+
+      stack[stack_size++] = StackElement(curr_col, 0);
+      //depth first search
+      while (stack_size > 0) {
+        StackElementPtr element = &stack[stack_size - 1];
+        curr_col = element->col;
+
+        if (curr_col.type().id() == cudf::type_id::STRUCT) {
+          if (element->child_idx == curr_col.num_child_columns()) {
+            // All child columns processed, pop the stack
+            stack_size--;
+          } else {
+            // Push child column to stack
+            stack[stack_size++] = StackElement(cudf::detail::structs_column_device_view(curr_col).get_sliced_child(element->child_idx), element->factor_exp + curr_col.num_child_columns() - 1 - element->child_idx);
```

Comment:
Add `throw exception` if `stack_size > 8`

Reply:
See an example as below, a mcro is used to handle this kind of case. And `CUDF_UNREACHABLE ` 
```
    template <typename T, CUDF_ENABLE_IF(not cudf::column_device_view::has_element_accessor<T>())>
    __device__ hash_value_type operator()(cudf::column_device_view const&,
                                          cudf::size_type,
                                          Nullate const,
                                          hash_value_type const) const noexcept
    {
      CUDF_UNREACHABLE("Unsupported type for xxhash64");
    }
```

Reply:
Throw an exception in CPU code before call into this kernel.